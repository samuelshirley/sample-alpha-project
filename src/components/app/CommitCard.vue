<template>
  <v-card>
    <v-toolbar
      flat
      card
    >
      <v-toolbar-side-icon
        @click="goBack"
      >
        <v-icon>mdi-chevron-left</v-icon>
      </v-toolbar-side-icon>
      <v-toolbar-title v-text="`Commit ${sha}`" />
    </v-toolbar>
    <v-card-text>
      <v-layout
        row
        align-start
        justify-center
      >
        <template v-if="Object.keys(details).length">
          <v-flex
            xs6
            text-xs-center
          >
            <v-avatar size="128px">
              <v-img
                v-if="Object.keys(details).length"
                :src="details.author.avatar_url"
              />
              <v-icon
                v-else
                v-text="'mdi-account'"
              />
            </v-avatar>
            <br>
            <div class="font-weight-bold title mt-3">
              {{ details.commit.author.name }}
            </div>
          </v-flex>
          <v-flex xs6>
            <v-layout
              wrap
              pb-3
            >
              <v-flex
                xs4
                text-xs-center
              >
                <span class="title">Total Additions</span>
              </v-flex>
              <v-flex
                xs4
                text-xs-center
              >
                <span class="title">Total Deletions</span>
              </v-flex>
              <v-flex
                xs4
                text-xs-center
              >
                <span class="title">Total Overall</span>
              </v-flex>
              <v-flex
                xs4
                text-xs-center
              >
                <span class="headline green--text">+{{ details.stats.additions }}</span>
              </v-flex>
              <v-flex
                xs4
                text-xs-center
              >
                <span class="headline red--text">-{{ details.stats.deletions }}</span>
              </v-flex>
              <v-flex
                xs4
                text-xs-center
              >
                <span class="headline blue--text">{{ details.stats.total }}</span>
              </v-flex>
            </v-layout>
            <v-subheader v-text="'All Files Changed'" />
            <div
              :style="{maxHeight: computedHeight}"
              style="overflow-x:hidden;"
              class="scroll-y"
            >
              <v-list>
                <v-list-tile
                  v-for="file in details.files"
                  :key="file.sha"
                >
                  <v-list-tile-content v-text="file.filename" />
                </v-list-tile>
              </v-list>
            </div>
          </v-flex>
        </template>
        <template v-else>
          <v-progress-circular
            :size="70"
            :width="5"
            color="primary"
            indeterminate
          />
        </template>
      </v-layout>
    </v-card-text>
  </v-card>
</template>

<script>
import { convertToUnit } from 'vuetify/es5/util/helpers'
import { mapActions } from 'vuex'

export default {
  props: {
    sha: {
      type: String,
      required: true,
    },
  },
  data () {
    return {
      details: {},
    }
  },
  computed: {
    computedHeight () {
      return convertToUnit('560px')
    },
  },
  async created () {
    this.loading = true
    try {
      this.details = await this.fetchCommit(this.sha)
    } catch (e) {
      this.$snackbar.error('Error fetching commits.')
    } finally {
      this.loading = false
    }
  },
  methods: {
    ...mapActions('github', [
      'fetchCommit',
    ]),
    goBack () {
      const query = {}
      if (this.$route.query.page) query.page = this.$route.query.page
      if (this.$route.query.per_page) query.per_page = this.$route.query.per_page
      this.$router.push({ path: '/commits', query })
    },
  },
}
</script>

<style>

</style>
