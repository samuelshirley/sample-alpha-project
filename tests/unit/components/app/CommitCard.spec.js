import Vuex from 'vuex'
import { shallowMount } from '@vue/test-utils'
import githubState from '@/store/modules/github/state'
import githubMutations from '@/store/modules/github/mutations'
import CommitCard from '@/components/app/CommitCard.vue'

const $snackbar = {
  error: jest.fn(),
}

describe('CommitCard.vue', () => {
  let store
  let fetchSpy = jest.fn().mockResolvedValue({
    'sha': '175b9acfd438be86de7687bc48bf9d84c0ee4949',
    'node_id': 'MDY6Q29tbWl0Njc5NjI2NDg6MTc1YjlhY2ZkNDM4YmU4NmRlNzY4N2JjNDhiZjlkODRjMGVlNDk0OQ==',
    'commit': {
      'author': {
        'name': 'MajesticPotatoe',
        'email': 'AMajesticPotatoe@gmail.com',
        'date': '2019-03-22T13:34:45Z',
      },
      'committer': {
        'name': 'MajesticPotatoe',
        'email': 'AMajesticPotatoe@gmail.com',
        'date': '2019-03-22T13:35:29Z',
      },
      'message': 'docs: update search-input prop desc\n\nCloses #6794',
      'url': 'https://api.github.com/repos/vuetifyjs/vuetify/git/commits/175b9acfd438be86de7687bc48bf9d84c0ee4949',
      'comment_count': 0,
      'verification': {
        'verified': false,
        'reason': 'unsigned',
        'signature': null,
        'payload': null,
      },
    },
    'author': {
      'login': 'MajesticPotatoe',
      'id': 27827979,
      'avatar_url': 'https://avatars1.githubusercontent.com/u/27827979?v=4',
      'type': 'User',
      'site_admin': false,
    },
    'committer': {
      'login': 'MajesticPotatoe',
      'id': 27827979,
      'node_id': 'MDQ6VXNlcjI3ODI3OTc5',
      'avatar_url': 'https://avatars1.githubusercontent.com/u/27827979?v=4',
      'type': 'User',
      'site_admin': false,
    },
    'stats': {
      'total': 2,
      'additions': 1,
      'deletions': 1,
    },
    'files': [
      {
        'sha': '266a65415a56f6d531b5571aa4a388896ba7923f',
        'filename': 'packages/docs/src/lang/en/components/Selects.json',
        'status': 'modified',
        'additions': 1,
        'deletions': 1,
        'changes': 2,
        'blob_url': 'https://github.com/vuetifyjs/vuetify/blob/175b9acfd438be86de7687bc48bf9d84c0ee4949/packages/docs/src/lang/en/components/Selects.json',
        'raw_url': 'https://github.com/vuetifyjs/vuetify/raw/175b9acfd438be86de7687bc48bf9d84c0ee4949/packages/docs/src/lang/en/components/Selects.json',
        'contents_url': 'https://api.github.com/repos/vuetifyjs/vuetify/contents/packages/docs/src/lang/en/components/Selects.json?ref=175b9acfd438be86de7687bc48bf9d84c0ee4949',
        'patch': '@@ -61,7 +61,7 @@\n     "openOnClear": "When using the **clearable** prop, once cleared, the select menu will either open or stay open, depending on the current state",\n     "overflow": "Creates an overflow button - [spec](https://material.io/guidelines/components/buttons.html#buttons-dropdown-buttons)",\n     "returnObject": "Changes the selection behavior to return the object directly rather than the value specified with **item-value**",\n-    "searchInput": "Bound when using the autocomplete prop. Use the **.sync** modifier to catch user input from the autocomplete search input",\n+    "searchInput": "Use the **.sync** modifier to catch user input from the search input",\n     "segmented": "Creates a segmented button - [spec](https://material.io/guidelines/components/buttons.html#buttons-dropdown-buttons)",\n     "smallChips": "Changes display of selections to chips with the **small** property",\n     "tags": "Tagging functionality, allows the user to create new values not available from the **items** prop",',
      },
    ],
  })
  beforeEach(() => {
    store = new Vuex.Store({
      modules: {
        github: {
          namespaced: true,
          state: githubState,
          mutations: githubMutations,
          actions: {
            fetchCommit: fetchSpy,
          },
        },
      },
    })
  })
  it('renders correctly when commit loaded', async () => {
    const wrapper = shallowMount(CommitCard, {
      propsData: {
        sha: '175b9acfd438be86de7687bc48bf9d84c0ee4949',
      },
      mocks: {
        $snackbar,
      },
      store,
    })
    await wrapper.vm.$nextTick()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders correctly when commit not loaded yet', async () => {
    const wrapper = shallowMount(CommitCard, {
      propsData: {
        sha: '175b9acfd438be86de7687bc48bf9d84c0ee4949',
      },
      mocks: {
        $snackbar,
      },
      store,
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
