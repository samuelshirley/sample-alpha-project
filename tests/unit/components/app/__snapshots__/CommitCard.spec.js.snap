// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommitCard.vue renders correctly when commit loaded 1`] = `
<v-card-stub tag="div">
  <v-toolbar-stub card="true" flat="true" scrollthreshold="300">
    <v-toolbar-side-icon-stub>
      <v-icon-stub>mdi-chevron-left</v-icon-stub>
    </v-toolbar-side-icon-stub>
    <v-toolbar-title-stub></v-toolbar-title-stub>
  </v-toolbar-stub>
  <v-card-text-stub>
    <v-layout-stub tag="div" row="" align-start="" justify-center="">
      <v-flex-stub tag="div" xs6="" text-xs-center="">
        <v-avatar-stub size="128px">
          <v-img-stub src="https://avatars1.githubusercontent.com/u/27827979?v=4" position="center center" transition="fade-transition"></v-img-stub>
        </v-avatar-stub> <br>
        <div class="font-weight-bold title mt-3">
          MajesticPotatoe
        </div>
      </v-flex-stub>
      <v-flex-stub tag="div" xs6="">
        <v-layout-stub tag="div" wrap="" pb-3="">
          <v-flex-stub tag="div" xs4="" text-xs-center=""><span class="title">Total Additions</span></v-flex-stub>
          <v-flex-stub tag="div" xs4="" text-xs-center=""><span class="title">Total Deletions</span></v-flex-stub>
          <v-flex-stub tag="div" xs4="" text-xs-center=""><span class="title">Total Overall</span></v-flex-stub>
          <v-flex-stub tag="div" xs4="" text-xs-center=""><span class="headline green--text">+1</span></v-flex-stub>
          <v-flex-stub tag="div" xs4="" text-xs-center=""><span class="headline red--text">-1</span></v-flex-stub>
          <v-flex-stub tag="div" xs4="" text-xs-center=""><span class="headline blue--text">2</span></v-flex-stub>
        </v-layout-stub>
        <v-subheader-stub>All Files Changed</v-subheader-stub>
        <div class="scroll-y" style="overflow-x: hidden; max-height: 560px;">
          <v-list-stub>
            <v-list-tile-stub activeclass="primary--text">
              <v-list-tile-content-stub></v-list-tile-content-stub>
            </v-list-tile-stub>
          </v-list-stub>
        </div>
      </v-flex-stub>
    </v-layout-stub>
  </v-card-text-stub>
</v-card-stub>
`;

exports[`CommitCard.vue renders correctly when commit not loaded yet 1`] = `
<v-card-stub tag="div">
  <v-toolbar-stub card="true" flat="true" scrollthreshold="300">
    <v-toolbar-side-icon-stub>
      <v-icon-stub>mdi-chevron-left</v-icon-stub>
    </v-toolbar-side-icon-stub>
    <v-toolbar-title-stub></v-toolbar-title-stub>
  </v-toolbar-stub>
  <v-card-text-stub>
    <v-layout-stub tag="div" row="" align-start="" justify-center="">
      <v-progress-circular-stub color="primary" indeterminate="true" rotate="0" size="70" width="5" value="0"></v-progress-circular-stub>
    </v-layout-stub>
  </v-card-text-stub>
</v-card-stub>
`;
